src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -760,6 +761,17 @@ private static boolean isVCFStream(final InputStream stream, final String MAGIC_
         return values;
     }
 
+    /**
+     * Parses a String as a Double, being tolerant for lower-case NaN (nan).
+     */
+    private static final Double decodeDouble(final String string) {
```

Comment:
I would expose this somewhere public and maybe rename it to be clearer how it's different from the standard methods.  Maybe `VCFUtils.parseDoubleAccordingToVcfSpec`.

Reply:
There are a number of additional places that this should be used as well I believe:

`GenotypeLikelihoods.parseDeprecatedGLString` line 270 parseDouble`
 `CommonInfo` 299 and 323 that use `Double.valueOf`
`VariantContext` 1632
`Genotype` 529